MNE/deep/deepeval.cpp

```diff
@@ -0,0 +1,217 @@
+
+#include "deepeval.h"
+
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// STL INCLUDES
+//=============================================================================================================
+
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// Qt INCLUDES
+//=============================================================================================================
+
+#include <QFile>
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// USED NAMESPACES
+//=============================================================================================================
+
+using namespace DEEPLIB;
+using namespace Microsoft::MSR::CNTK;
+
+
+// Used for retrieving the model appropriate for the element type (float / double)
+template<typename ElemType>
+using GetEvalProc = void(*)(IEvaluateModel<ElemType>**);
+
+
+typedef std::pair<std::wstring, std::vector<float>*> MapEntry;
+typedef std::map<std::wstring, std::vector<float>*> Layer;
+
+
+//*************************************************************************************************************
+//=============================================================================================================
+// DEFINE MEMBER METHODS
+//=============================================================================================================
+
+DeepEval::DeepEval()
+: m_model(NULL)
+{
+}
+
+
+//*************************************************************************************************************
+
+DeepEval::DeepEval(const QString &sModelFilename)
+: m_sModelFilename(sModelFilename)
+, m_model(NULL)
+{
+    loadModel();
+}
+
+
+//*************************************************************************************************************
+
+DeepEval::~DeepEval()
+{
+
+}
+
+
+//*************************************************************************************************************
+
+const QString& DeepEval::getModelFilename() const
+{
+    return m_sModelFilename;
+}
+
+
+//*************************************************************************************************************
+
+void DeepEval::setModelFilename(const QString &sModelFilename)
+{
+    m_sModelFilename = sModelFilename;
+}
+
+
+//*************************************************************************************************************
+
+bool DeepEval::evalModel(std::vector<float>& inputs, std::vector<float>& outputs)
```

Comment:
i'm afraid a CNTK method used inside doe only take non const - thats the reason while it is here also not const